File: nodejs_backend/models/subject.model.js
const mongoose = require('mongoose');
const { Schema } = mongoose;
const db = require('../config/db');

const subjectSchema = new Schema({
    subjectID: {
        type: String,
        required: true,
        unique: true,
    },
    subjectName: {
        type: String,
        required: true,
    },
    numberOfCredits: {
        type: Number,
        required: true,
    },
    numberOfLessons: {
        type: Number,
        required: true,
    },
}, { timestamps: true });

const Subject = db.model('subject', subjectSchema);
module.exports = Subject;

File: flutter/lib/model/subjectModel.dart
class Subject {
  final String id;
  final String subjectID;
  final String subjectName;
  final int numberOfCredits;
  final int numberOfLessons;

  Subject(
      {required this.id,
      required this.subjectID,
      required this.subjectName,
      required this.numberOfCredits,
      required this.numberOfLessons});

  factory Subject.fromMap(Map<String, dynamic> json) {
    return Subject(
      id: json['_id'],
      subjectID: json['subjectID'],
      subjectName: json['subjectName'],
      numberOfCredits: json['numberOfCredits'],
      numberOfLessons: json['numberOfLessons'],
    );
  }

  @override
  String toString() {
    return '${id} - ${subjectID} - ${subjectName} - ${numberOfCredits} - ${numberOfLessons}';
  }
}
